watertap/examples/flowsheets/case_studies/electroNP/electroNP_flowsheet_extended.py

```diff
@@ -0,0 +1,498 @@
+    m.fs.CL1.split_fraction[0, "effluent", "S_NO3"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_O2"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_PO4"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_IC"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_K"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "S_Mg"].fix(0.49986)
+    m.fs.CL1.split_fraction[0, "effluent", "X_AUT"].fix(0.022117)
+    m.fs.CL1.split_fraction[0, "effluent", "X_H"].fix(0.021922)
+    m.fs.CL1.split_fraction[0, "effluent", "X_I"].fix(0.021715)
+    m.fs.CL1.split_fraction[0, "effluent", "X_PAO"].fix(0.022)
+    m.fs.CL1.split_fraction[0, "effluent", "X_PHA"].fix(0.02147)
+    m.fs.CL1.split_fraction[0, "effluent", "X_PP"].fix(0.02144)
+    m.fs.CL1.split_fraction[0, "effluent", "X_S"].fix(0.02221)
+
+    # Sludge purge separator
+    m.fs.SP6.split_fraction[:, "recycle"].fix(0.65)
+
+    # AD
+    m.fs.AD.volume_liquid.fix(3400)
+    m.fs.AD.volume_vapor.fix(300)
+    m.fs.AD.liquid_outlet.temperature.fix(298.15)
+
+    # ElectroNP
+    m.fs.electroNP.energy_electric_flow_mass.fix(0.044 * pyunits.kWh / pyunits.kg)
+    m.fs.electroNP.magnesium_chloride_dosage.fix(0.388)
+
+    # Costing
+    m.fs.costing.electroNP.phosphorus_recovery_value = 0
+
+    # Check degrees of freedom
+    print(degrees_of_freedom(m))
+    assert degrees_of_freedom(m) == 0
+
+    def scale_variables(m):
+        for var in m.fs.component_data_objects(pyo.Var, descend_into=True):
+            if "flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e1)
+            if "thickener.properties_in[0.0].flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e5)
+            if "translator_asm2d_adm1.properties_in[0.0].flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e5)
+            if "AD.liquid_phase.properties_in[0.0].flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e5)
+            if "translator_adm1_asm2d.properties_in[0.0].flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e5)
+            if "dewater.properties_in[0.0].flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e5)
+            if "electroNP.mixed_state[0.0].flow_vol" in var.name:
+                iscale.set_scaling_factor(var, 1e5)
+            if "temperature" in var.name:
+                iscale.set_scaling_factor(var, 1e-1)
+            if "pressure" in var.name:
+                iscale.set_scaling_factor(var, 1e-4)
+
+    scale_variables(m)
+    iscale.calculate_scaling_factors(m.fs)
+
+    # Initialize flowsheet
+    # Apply sequential decomposition - 1 iteration should suffice
+    seq = SequentialDecomposition()
+    seq.options.select_tear_method = "heuristic"
+    seq.options.tear_method = "Wegstein"
+    seq.options.iterLim = 0
+
+    def function(unit):
+        unit.initialize(outlvl=idaeslog.INFO, optarg={"bound_push": 1e-2})
```

Comment:
```suggestion
        unit.initialize(outlvl=idaeslog.INFO)
```
Hmmm...this still doesn't look like you're using the latest form of the flowsheet since this was removed. I threw up a PR for the full flowsheet (based on latest BSM2 flowsheet) to supplement your efforts here and see if I can arrive at a solution for at least the full flowsheet without electronP

Reply:
@adam-a-a, it does not matter the default value for bound_push is `1e-2`, so actually setting at 1e-2 does not change anything

Reply:
I know this doesn't change anything because it's the default, but this was still not included in the latest flowsheet implementation which I thought you were trying to blend with electronP, with some simplifications regarding recycle loops. So it looked like you were still using an old version. Maybe that isn't the case.

Reply:
The major difference is I need to include `seq.options.select_tear_method = "heuristic"` for successfully running the flowsheet, but the latest one does not have this line. I don't include tear guess because it does not have a loop for this flowsheet.